.github/workflows/build_and_test.yml

```diff
@@ -790,7 +790,8 @@ jobs:
 
   maven-build:
     needs: precondition
-    if: fromJson(needs.precondition.outputs.required).maven-build == 'true'
+    # Skipped in branch-3.4, see SPARK-47090.
+    if: fromJson(needs.precondition.outputs.required).maven-build == 'true' && inputs.branch != 'branch-3.4'
```

Comment:
I can add more condition, e.g., only skip JDK 21 in branch-3.5 and branch-3.4 if that's preferred

Reply:
cc @dongjoon-hyun do you have any preference on this?

Reply:
Perhaps we can use a separate job to generate the content of `java-matrix` first? Although this might increase complexity

Reply:
Yeah we can do something more fine grained. I prefer to just take it out in branch-3.4 because compilation itself doesn't affect the Spark 3.4 releases.

Reply:
hmm... I think after https://github.com/apache/spark/pull/45148/files, the daily job for branch-3.4 will no longer execute maven-build ...

Reply:
https://github.com/apache/spark/blob/c4e4497ff7e747eb71d087cdfb1b51673c53b83b/.github/workflows/build_branch34.yml#L43-L51

For daily build, `maven-build` is not true,  before https://github.com/apache/spark/pull/45148, the `apple-silicon` job was checking the value of `build`

Reply:
https://github.com/apache/spark/actions/runs/7957014021

Reply:
Ohh okie that's good then :-). Let me close this